cvmfs/sync_mediator.cc

```diff
@@ -93,6 +94,30 @@ void SyncMediator::Add(SharedPtr<SyncItem> entry) {
     AddDirectoryRecursively(entry);
     return;
   }
+  
+  // .cvmfsbundles file type
+  if(entry->isBundleSpec()){
+    printf("cvmfsbundles file found. filename: %s\n", (entry->GetRelativePath()).c_str());
+
+    std::string json_string;
+
+    int fd = open (entry->GetUnionPath().c_str(), O_RDONLY);
+    if(fd >= 0) {
+      if(SafeReadToString(fd, &json_string)) {
```

Comment:
Same here, let's PANIC if reading fails.
```suggestion
      if (SafeReadToString(fd, &json_string)) {
```